pkgs/cupertino_http/example/integration_test/url_session_configuration_test.dart

```diff
@@ -2,10 +2,39 @@
 // for details. All rights reserved. Use of this source code is governed by a
 // BSD-style license that can be found in the LICENSE file.
 
+import 'dart:io';
+
 import 'package:cupertino_http/cupertino_http.dart';
 import 'package:integration_test/integration_test.dart';
 import 'package:test/test.dart';
 
+/// Make a HTTP request using the given configuration and return the headers
+/// received by the server.
+Future<Map<String, List<String>>> sentHeaders(
+    URLSessionConfiguration config) async {
+  final session = URLSession.sessionWithConfiguration(config);
+  final headers = <String, List<String>>{};
+  final server = (await HttpServer.bind('localhost', 0))
+    ..listen((request) async {
+      request.headers.forEach((k, v) => headers[k] = v);
+      await request.drain<void>();
+      request.response.headers.set('Content-Type', 'text/plain');
+      request.response.write('Hello World');
+      await request.response.close();
+    });
+
+  final task = session.dataTaskWithRequest(
+      URLRequest.fromUrl(Uri.parse('http://localhost:${server.port}')))
+    ..resume();
```

Comment:
[nit] I generally prefer to avoid writing URL strings directly.

We unfortunately don't have a constructor that maps nicely when we have to include the port, but even with a manually concatenated authority string I think it's an improvement over `Uri.parse`

```suggestion
  final task = session.dataTaskWithRequest(URLRequest.fromUrl(
      Uri(scheme: 'http', host: 'localhost', port: server.port)))
    ..resume();
```

```suggestion
  final task = session.dataTaskWithRequest(
      URLRequest.fromUrl(Uri.http('localhost:${server.port}')))
    ..resume();
```